sky/packages/sky/lib/src/widgets/basic.dart

```diff
@@ -145,7 +146,7 @@ class CustomPaint extends OneChildRenderObjectWidget {
   RenderCustomPaint createRenderObject() => new RenderCustomPaint(onPaint: onPaint);
 
   void updateRenderObject(RenderCustomPaint renderObject, CustomPaint oldWidget) {
-    if (oldWidget != null && oldWidget.token != token)
+    if (oldWidget.token != token)
       renderObject.markNeedsPaint();
     renderObject.onPaint = onPaint;
   }
```

Comment:
you should also document the issue about the callback changing each frame causing a repaint each frame